homeassistant/components/schlage/coordinator.py

```diff
@@ -32,10 +42,29 @@ def __init__(self, hass: HomeAssistant, username: str, api: Schlage) -> None:
     async def _async_update_data(self) -> SchlageData:
         """Fetch the latest data from the Schlage API."""
         try:
-            return await self.hass.async_add_executor_job(self._update_data)
-        except Error as ex:
+            locks: list[Lock] = await self.hass.async_add_executor_job(self.api.locks)
+        except SchlageError as ex:
             raise UpdateFailed("Failed to refresh Schlage data") from ex
+        lock_data = await asyncio.gather(
+            *(
+                self.hass.async_add_executor_job(self._get_lock_data, lock)
+                for lock in locks
+            )
+        )
+        return SchlageData(
+            locks={ld.lock.device_id: ld for ld in lock_data},
+        )
 
-    def _update_data(self) -> SchlageData:
-        """Fetch the latest data from the Schlage API."""
-        return SchlageData(locks={lock.device_id: lock for lock in self.api.locks()})
+    def _get_lock_data(self, lock: Lock) -> LockData:
+        logs: list[LockLog] = []
+        previous_lock_data = None
+        if self.data and (previous_lock_data := self.data.locks.get(lock.device_id)):
+            # Default to the previous data, in case a refresh fails.
+            # It's not critical if we don't have the freshest data.
+            logs = previous_lock_data.logs
+        try:
+            logs = lock.logs()
+        except SchlageError as ex:
+            LOGGER.warning('Failed to read logs for lock "%s": %s', lock.name, ex)
```

Comment:
Drop this to debug or only log once at warning level and than drop to debug as we don't want it to log every time it poll